storm-core/src/jvm/org/apache/storm/command/AdminCommands.java

```diff
@@ -0,0 +1,114 @@
+    private static Map conf;
+
+    public static void main(String [] args) throws Exception {
+
+        if (args.length == 0) {
+            throw new IllegalArgumentException("Missing command.");
+        }
+        initialize();
+        String command = args[0];
+        String[] newArgs = Arrays.copyOfRange(args, 1, args.length);
+        switch (command) {
+            case "remove_corrupt_topologies":
+                removeCorruptTopologies();
+                break;
+            default:
+                throw new RuntimeException("" + command + " is not a supported admin command");
+        }
+
+    }
+
+    private static void initialize() {
+        Map conf = ConfigUtils.readStormConfig();
+        ClientBlobStore clientBlobStore = Utils.getClientBlobStore(conf);
+        List<String> servers = (List<String>) conf.get(Config.STORM_ZOOKEEPER_SERVERS);
+        Object port = conf.get(Config.STORM_ZOOKEEPER_PORT);
+        List<ACL> acls = null;
+        if (Utils.isZkAuthenticationConfiguredStormServer(conf)) {
+            acls = adminZkAcls();
+        }
+        try {
+            IStormClusterState stormClusterState = ClusterUtils.mkStormClusterState(conf, acls, new ClusterStateContext(DaemonType.UNKNOWN));
+        } catch (Exception e) {
+            LOG.error("admin can't create stormClusterState");
+            throw Utils.wrapInRuntime(e);
+        }
+        CuratorFramework zk = Zookeeper.mkClient(conf, servers, port, "", new DefaultWatcherCallBack(),conf);
+    }
+
+    // we might think of moving this method in Utils class
+    private static List<ACL> adminZkAcls() {
+        final List<ACL> acls = new ArrayList<>();
+        acls.add(ZooDefs.Ids.CREATOR_ALL_ACL.get(0));
+        acls.add(new ACL((ZooDefs.Perms.READ ^ ZooDefs.Perms.CREATE), ZooDefs.Ids.ANYONE_ID_UNSAFE));
+        return acls;
+    }
+
+    private static void removeCorruptTopologies( ) {
+        Iterator<String> corruptTopologies = listCorruptTopologies();
+        while(corruptTopologies.hasNext()) {
+            stormClusterState.removeStorm(corruptTopologies.next());
+        }
+    }
+
+    private static Iterator<String> listCorruptTopologies() {
+        Iterator<String> blobStoreTopologyIds = clientBlobStore.listKeys();
+        Set<String> activeTopologyIds = new HashSet<>(Zookeeper.getChildren(zk, conf.get(Config.STORM_ZOOKEEPER_ROOT) + ClusterUtils.STORMS_SUBTREE, false));
+        HashSet<String> blobTopologyIds = Sets.newHashSet(blobStoreTopologyIds);
+        Sets.SetView<String> diffTopology = Sets.difference(activeTopologyIds, blobTopologyIds);
+        LOG.info("active-topology-ids [{}] blob-topology-ids [{}] diff-topology [{}]",
+                generateJoinedString(activeTopologyIds), generateJoinedString(blobTopologyIds),
+                generateJoinedString(diffTopology));
+        return diffTopology.iterator();
+    }
+
+    private static String generateJoinedString(Set<String> activeTopologyIds) {
```

Comment:
I'm not sure we need this.  toString of most sets already return this information.